src/extensions/default/NoDistractions/main.js

```diff
@@ -0,0 +1,155 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, brackets */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Menus               = brackets.getModule("command/Menus"),
+        CommandManager      = brackets.getModule("command/CommandManager"),
+        Commands            = brackets.getModule("command/Commands"),
+        Strings             = brackets.getModule("strings"),
+        PreferencesManager  = brackets.getModule("preferences/PreferencesManager"),
+        ViewUtils           = brackets.getModule("utils/ViewUtils"),
+        KeyBindingManager   = brackets.getModule("command/KeyBindingManager"),
+        WorkspaceManager    = brackets.getModule("view/WorkspaceManager");
+
+    // Constants
+    var PREFS_PURE_CODE           = "noDistractions",
+        CMD_TOGGLE_PURE_CODE      = "view.togglePureCode",
+        CMD_TOGGLE_PANELS         = "view.togglePanels";
+
+    //key binding keys
+    var togglePureCodeKey         = "Ctrl-Shift-2",
+        togglePureCodeKeyMac      = "Cmd-Shift-2",
+        togglePanelsKey           = "Ctrl-Shift-`",
+        togglePanelsKeyMac        = "Cmd-Shift-`";
+
+    //locals
+    var _previouslyOpenPanelIDs = [],
+        panelsToggled = false,
+        layoutUpdated = false;
+
+    /**
+     * @private
+     *
+     * Updates the command checked status based on the preference name given.
+     *
+     * @param {string} name Name of preference that has changed
+     */
+    function _updateCheckedState() {
+        CommandManager.get(CMD_TOGGLE_PURE_CODE).setChecked(PreferencesManager.get(PREFS_PURE_CODE));
+    }
+
+    function _togglePureCode() {
+        PreferencesManager.set(PREFS_PURE_CODE, !PreferencesManager.get(PREFS_PURE_CODE));
+    }
+
+    /**
+     * hide all open panels
+     */
+    function _hidePanelsIfRequired() {
+        var panelIDs = WorkspaceManager.getAllPanelIDs(), i = 0;
+        _previouslyOpenPanelIDs = [];
+        for (i = 0; i < panelIDs.length; i++) {
+            if (WorkspaceManager.getPanelForID(panelIDs[i]).isVisible()) {
+                WorkspaceManager.getPanelForID(panelIDs[i]).hide();
+                _previouslyOpenPanelIDs.push(panelIDs[i]);
+            }
+        }
+    }
+
+    /**
+     * show all open panels that was previously hidden by _hidePanelsIfRequired()
+     */
+    function _showPanelsIfRequired() {
+        var panelIDs = _previouslyOpenPanelIDs, i = 0;
+        for (i = 0; i < panelIDs.length; i++) {
+            if (WorkspaceManager.getPanelForID(panelIDs[i])) {
```

Comment:
Store the result from `getPanelForID` instead of calling the same function below.